app/renderer/src/reducers/keys.js

```diff
@@ -0,0 +1,17 @@
+import { SET_ACTIVE_KEY, CLEAR_ACTIVE_KEY } from '../actions/types';
+
+const initialState = [];
+
+export default function (state = initialState, action) {
+  switch (action.type) {
+    case SET_ACTIVE_KEY:
+      return state.concat([action.key]);
+    case CLEAR_ACTIVE_KEY:
+      return [
+        ...state.slice(0, state.indexOf(action.key)),
+        ...state.slice(state.indexOf(action.key) + 1),
+      ];
```

Comment:
This is a question of preference now, but I really dislike this. Readability is hard and there 2 things happening at the same time on respective lines..

Since we are spreading to ensure a __copy__ of the state being returned, why is `splice` not taken into consideration here?

```
const clearedState = state.splice(/**... no calculation */)
return [ ...clearedState ]
```